### DISCUSSION TITLE
Avalonia.Lottie after DrawingContext refactoring

### DISCUSSION THREAD

**Discussion by msneijders** [AvaloniaUI/Avalonia#10563] [Q&A]
Using '11.0.999-cibuild0029495-beta' I had Avalonia.Lottie running -- now after a recent refactoring of `DrawingContext` (now trying out '11.0.999-cibuild0031494-beta') I have some trouble getting `Avalonia.Lottie.Animation.Content.LottieCanvas` to compile.

LottieCanvas uses a `Stack<DrawingContext>` and instantiates `DrawingContext` using a `IDrawingContextImpl` -- but I think this is refactored to a `PlatformDrawingContext` marked `internal`, so it cannot do this anymore.

I am now trying to get it back working using the `IDrawingContextImpl` itself -- would this be the right way?

**Comment by kekekeks** [comment]
Check if `ImmediateDrawingContext` is sufficient for lottie needs

**Reply by msneijders** [reply]
I will try this, as soon as I get the avalonia xaml name reference generator working again, which also seem to have changed (but only setting 'IncludeAvaloniaGenerators' to true in csproj does not seem to trigger it)

**Reply by msneijders** [reply]
Had to get rid of that source generator, as I could not get it to work -- but got `ImmediateDrawingContext` tested in `LottieCanvas` -- refactored `ContextStack` to `Stack<ImmediateDrawingContext>` and used `CurrentDrawingContext.PlatformImpl` to use `DrawGeometry(..)` -- and this works!

Still I had to use `reflection` to get an `ImmediateDrawingContext` as the  constructor is `internal`; this does not feel like a very stable solution.
